### PR TITLE
Add end to end tests, with real server and client

### PR DESCRIPTION
**Pull request by rojer9-fb**
Client and server binary are refactored to be reusable within the same process,
tests/e2e/e2e_test starts server on a random port, uses client to submit jobs and check status.
This is as close to using standalone binaries as possible but still provides coverage.

Two E2E tests are implemented:
 * A simple test that runs a single job in one go and checks its status.
 * A pause/resume test that starts a job and then stops server multiple
   times to force pause and resume, then verifies that all teh steps
   have been executed once and in the correct order.

Signed-off-by: Deomid "rojer" Ryabkov <rojer9@fb.com>

### CODE REVIEW

**Review by tfg13 — CHANGES_REQUESTED** [review]
Is the 7MB compiled client binary included on purpose? That will bloat the repo

File: plugins/storage/rdbms/init.go [hunk 1]
```diff
@@ -124,6 +124,16 @@ func (r *RDBMS) Version() (uint64, error) {
 	return migrationlib.DBVersion(r.db)
 }
 
+// Reset wipes entire database contents. Used in tests.
+func (r *RDBMS) Reset() error {
```
Comment (tfg13) [inline plugins/storage/rdbms/init.go:128]: Scary stuff. Should we at least yell a bit, like https://github.com/facebookincubator/contest/blob/master/plugins/targetlocker/dblocker/dblocker.go#L343 ?

Also DELETE vs. TRUNCATE. (not sure, just wanted to mention it)

File: pkg/jobmanager/jobmanager.go [hunk 2]
```diff
@@ -182,6 +183,7 @@ loop:
 	}
 	// Stop the API (if not already)
 	jm.StopAPI()
+	<-errCh
```
Comment (tfg13) [inline pkg/jobmanager/jobmanager.go:186]: This effectively waits for API shutdown, is that right?

Comment (rojer9-fb) [reply pkg/jobmanager/jobmanager.go:186]: correct

File: plugins/storage/rdbms/init.go (hunk 1 above)
Comment (rojer9-fb) [reply plugins/storage/rdbms/init.go:128]: there is no context to yell to though, the storage api doesn't use context (yet).

**Comment by rojer9-fb** [comment]
no, binary got there by accident, updated .gitignore to exclude.

File: plugins/storage/rdbms/init.go (hunk 1 above)
Comment (rojer9-fb) [reply plugins/storage/rdbms/init.go:128]: changed to TRUNCATE

**Comment by rojer9-fb** [comment]
PTAL

**Review by tfg13 — APPROVED** [review]

**Pull request merged**